### ISSUE TITLE
Upgrade to 0.15.2 (from 0.15.1) resultts in go trimmed error

### ISSUE THREAD

**Issue by justinrush** [gomods/athens#1990]
**Describe the bug**

We upgraded from `0.15.1` to `0.15.2` and now a simple:

```sh
go get github.com/nats-io/nats.go@latest
```

Results in a `404`.

**Error Message**

 The athens error logs show:

```
INFO[4:49PM]: exit status 2: go: cannot find GOROOT directory: 'go' binary is trimmed and GOROOT is not set
    http-method=GET http-path=/github.com/nats-io/@v/list kind=Not Found module= operation=download.ListHandler ops=[download.ListHandler pool.List protocol.List vcsLister.List] request-id=67210d5e-9557-4251-8661-1165b9eb8fb5 version= 
INFO[4:49PM]: incoming request    http-method=GET http-path=/github.com/nats-io/@v/list http-status=404 request-id=67210d5e-9557-4251-8661-1165b9eb8fb5 
INFO[4:49PM]: exit status 2: go: cannot find GOROOT directory: 'go' binary is trimmed and GOROOT is not set
    http-method=GET http-path=/github.com/nats-io/nats.go/@v/list kind=Not Found module= operation=download.ListHandler ops=[download.ListHandler pool.List protocol.List vcsLister.List] request-id=ee1585b9-5125-410c-8257-482d8365fdf0 version= 
INFO[4:49PM]: incoming request    http-method=GET http-path=/github.com/nats-io/nats.go/@v/list http-status=404 request-id=ee1585b9-5125-410c-8257-482d8365fdf0 
INFO[4:49PM]: exit status 2: go: cannot find GOROOT directory: 'go' binary is trimmed and GOROOT is not set
    http-method=GET http-path=/github.com/@v/list kind=Not Found module= operation=download.ListHandler ops=[download.ListHandler pool.List protocol.List vcsLister.List] request-id=6b3a1320-c0e7-4c3c-9510-fcc57efacfc6 version= 
INFO[4:49PM]: incoming request    http-method=GET http-path=/github.com/@v/list http-status=404 request-id=6b3a1320-c0e7-4c3c-9510-fcc57efacfc6 
```

**To Reproduce**

We downgraded to 0.15.1 and the problem went away. When we re-upgraded, the problem came back.

**Expected behavior**

no errors.

**Environment (please complete the following information):**
 - OS: linux amd64 on Kubernetes
 - Go version :  1.22.2
 - Proxy version : 0.15.2
 - Storage (fs/mongodb/s3 etc.) : AKS PVC

**Additional context**

Our env is set to:

```yaml
   - env:
     - name: ATHENS_GO_BINARY_ENV_VARS
       value: GOPRIVATE=go.ourprivatehost.com
     - name: ATHENS_DISK_STORAGE_ROOT
       value: /var/lib/athens
     - name: ATHENS_STORAGE_TYPE
       value: disk
     - name: ATHENS_GOGET_WORKERS
       value: "6"
     - name: ATHENS_GLOBAL_ENDPOINT
       value: https://proxy.golang.org
```

It looks like the new image `0.15.2` doesn't have `GOROOT` set but `0.15.1` does... running this fails for example:

```
$ nerdctl run --rm -it gomods/athens:v0.15.2 go version
go: cannot find GOROOT directory: 'go' binary is trimmed and GOROOT is not set
Error: exit status 2
```

I tried setting GOROOT manually to what 0.15.1 had (`/usr/local/go`). that. resolves the error seen with `go version`, but not the issues with athens itself.

**Comment by c-seeger** [comment]
same issue here

**Comment by tbnguyen1407** [comment]
Any update on this?

**Comment by matt0x6F** [comment]
Fixed in v0.15.3 with #1989. Duplicate of #1992
